jina/peapods/zmq/__init__.py

```diff
@@ -166,31 +174,29 @@ def _init_sockets(self) -> Tuple:
             self.logger.debug(
                 f'input {self.args.host_in}:{colored(self.args.port_in, "yellow")}'
             )
-
-            out_sock, out_addr = _init_socket(
-                ctx,
-                self.args.host_out,
-                self.args.port_out,
-                self.args.socket_out,
-                self.identity,
-                ssh_server=self.args.ssh_server,
-                ssh_keyfile=self.args.ssh_keyfile,
-                ssh_password=self.args.ssh_password,
-            )
-            self.logger.debug(
-                f'output {self.args.host_out}:{colored(self.args.port_out, "yellow")}'
-            )
+            if not self.args.dynamic_out_routing:
+                out_sock, out_addr = _init_socket(
+                    ctx,
+                    self.args.host_out,
+                    self.args.port_out,
+                    self.args.socket_out,
+                    self.identity,
+                    ssh_server=self.args.ssh_server,
+                    ssh_keyfile=self.args.ssh_keyfile,
+                    ssh_password=self.args.ssh_password,
+                )
+                self.logger.debug(
+                    f'output {self.args.host_out}:{colored(self.args.port_out, "yellow")}'
+                )
+            else:
```

Comment:
I like better to have `out_sock, out_addr = None, None` above as part of initialization better than in an `else`

Reply:
But this is the functional approach: Set each variable only once and don't overwrite it. Why would you initialize the variable and then overwrite them? I guess it boils down to C-style code vs functional programming (Scala?) style code.

Reply:
ok, I would consider it part of rhe initialization but fair enough